### PR TITLE
Support GetClusterShardingStats query in Typed #24466

### PR DESCRIPTION
**Pull request by patriknw**
References #24466

### CODE REVIEW

File: akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/ClusterShardingStatsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        pongProbe.receiveMessage()
+      }
+      enterBarrier("sharding-initialized")
+
+      runOn(first, second, third) {
+        val replyToProbe = TestProbe[ClusterShardingStats]()
+        sharding.shardState ! GetClusterShardingStats(typeKey, queryTimeout, replyToProbe.ref)
+
+        val stats = replyToProbe.receiveMessage(queryTimeout)
+        println(s"# stats: $stats") // FIXME
```
Comment (johanandren) [inline akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/ClusterShardingStatsSpec.scala:98]: runaway println

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingQuery.scala [hunk 2]
```diff
@@ -30,4 +39,33 @@ final case class GetShardRegionState(entityTypeKey: EntityTypeKey[_], replyTo: A
...
+    timeout: FiniteDuration,
+    replyTo: ActorRef[ClusterShardingStats])
+    extends ClusterShardingQuery {
+
+  /**
+   * Java API
+   *
+   * Query the statistics about the currently running sharded entities in the
+   * entire cluster. If the given `timeout` is reached without answers from all
+   * shard regions the reply will contain an empty map of regions.
```
Comment (johanandren) [inline akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingQuery.scala:64]: testing caveat here as well

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/ClusterShardingStatsSpec.scala (hunk 1 above)
Comment (patriknw) [reply akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/typed/ClusterShardingStatsSpec.scala]: thanks

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingQuery.scala (hunk 2 above)
Comment (patriknw) [reply akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingQuery.scala:64]: this is the constructor, I think it's enough have that note on the class level

**Review by chbatey — APPROVED** [review]
LGTM

**Comment by chbatey** [comment]
If you squash the println commit maybe merge this rather than squash merge?

**Comment by patriknw** [comment]
fixed commits, let's merge and NOT squash

**Pull request merged**